rcgen/src/csr.rs

```diff
@@ -115,6 +115,41 @@ impl CertificateSigningRequestParams {
 								.push(SanType::try_from_general(name)?);
 						}
 					},
+					x509_parser::extensions::ParsedExtension::ExtendedKeyUsage(eku) => {
+						if eku.any {
+							params.insert_extended_key_usage(crate::ExtendedKeyUsagePurpose::Any);
+						}
+						if eku.server_auth {
+							params.insert_extended_key_usage(
+								crate::ExtendedKeyUsagePurpose::ServerAuth,
+							);
+						}
+						if eku.client_auth {
+							params.insert_extended_key_usage(
+								crate::ExtendedKeyUsagePurpose::ClientAuth,
+							);
+						}
+						if eku.code_signing {
+							params.insert_extended_key_usage(
+								crate::ExtendedKeyUsagePurpose::CodeSigning,
+							);
+						}
+						if eku.email_protection {
+							params.insert_extended_key_usage(
+								crate::ExtendedKeyUsagePurpose::EmailProtection,
+							);
+						}
+						if eku.time_stamping {
+							params.insert_extended_key_usage(
+								crate::ExtendedKeyUsagePurpose::TimeStamping,
+							);
+						}
+						if eku.ocsp_signing {
+							params.insert_extended_key_usage(
+								crate::ExtendedKeyUsagePurpose::OcspSigning,
+							);
+						}
```

Comment:
I think this will silently drop instances of `x509_parser::ExtendedKeyUsage.other` holding `Vec<Oid<'a>>`.

Since there's no equivalent for an arbitrary EKU in rcgen I think it would be prudent to return `Err(Error::UnsupportedExtension)` if `!eku.other.is_empty()`

Reply:
Oh that makes sense